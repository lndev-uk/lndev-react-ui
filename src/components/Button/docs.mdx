import { Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Button from './';

# Button

UI Variants example

<Preview>
  <Story id="components-button--example" />
</Preview>
<Props of={Button} />

## UX Variants

Button can be use in different variants styles based on de **UX** System

<Preview>
  <Story id="components-button--variants" />
</Preview>

## Sizing

Overriding Inherint sizing.

<Preview>
  <Story id="components-button--sizes" />
</Preview>

## Mobile Full

Using the canvas tab you can change the viewport. On Tablet and mobile the with of the button will become 100% of parent container

<Preview>
  <Story id="components-button--mobile-full" />
</Preview>
