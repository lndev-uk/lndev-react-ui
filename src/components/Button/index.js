//@flow
import React from 'react';
import PropTypes from 'prop-types';

type ButtonType =
  | 'primary'
  | 'secondary'
  | 'green'
  | 'red'
  | 'outlined'
  | 'transparent';

type ButtonSize = 'big' | 'small';
type HtmlButtonType = 'button' | 'submit' | 'reset';

export type ButtonProps = {
  mobileFull: Boolean,
  label: String,
  size: ButtonSize,
  type: ButtonType,
  htmlType: HtmlButtonType,
  name: String,
  disabled: Boolean
};

export default function Button(props: ButtonProps): React.ReactNode {
  const { type, label, size, mobileFull, htmlType, name, disabled } = props;
  const sizeClassName = size ? 'button--' + size : '';
  const typeClassName = type ? 'button--' + type : '';
  const mobileFullClassName = mobileFull ? 'button--mobileFull' : '';
  const nameAttr = name ? { name } : {};
  const disabledAttr = disabled ? { disabled: 'disabled' } : {};
  return (
    <button
      type={htmlType}
      {...nameAttr}
      {...disabledAttr}
      className={`button  ${typeClassName} ${sizeClassName} ${mobileFullClassName}`}
    >
      {label}
    </button>
  );
}

Button.defaultProps = {
  disabled: false
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  mobileFull: PropTypes.bool,
  size: PropTypes.oneOf(['big', 'small']),
  type: PropTypes.oneOf([
    'primary',
    'secondary',
    'green',
    'red',
    'outlined',
    'transparent'
  ]),
  htmlType: PropTypes.oneOf(['button', 'submit', 'reset']),
  name: PropTypes.string,
  disabled: PropTypes.bool
};
