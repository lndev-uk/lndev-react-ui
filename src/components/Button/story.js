import React from 'react';
import Button from '.';

export default { title: 'Button' };

export const Types = () => (
  <div>
    <div>
      <Button type="primary" label="Primary" />
    </div>
    <div>
      <Button type="secondary" label="Secondary" />
    </div>
    <div>
      <Button type="green" label="green" />
    </div>
    <div>
      <Button type="red" label="red" />
    </div>
    <div>
      <Button type="outlined" label="outlined" />
    </div>
    <div>
      <Button type="transparent" label="transparent" />
    </div>
  </div>
);

export const Sizes = () => (
  <div>
    <div>
      <Button size="small" label="Small" />
    </div>
    <div>
      <Button label="Default" />
    </div>
    <div>
      <Button size="big" label="Big" />
    </div>
  </div>
);

export const mobileFull = () => (
  <div>
    <div>Enable Chrome Dev Tools and simulate mobile/tabelt</div>
    <div>
      <Button mobileFull={true} label="Full Width Mobile" />
    </div>
  </div>
);
