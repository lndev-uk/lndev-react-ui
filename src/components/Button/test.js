import React from 'react';
import { render } from '@testing-library/react';
import Button from '.';

describe('Button', () => {
  it('matches snapshot success', () => {
    const { container } = render(<Button type="primary" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot secondary', () => {
    const { container } = render(
      <Button type="secondary" label="test label" />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot green', () => {
    const { container } = render(<Button type="green" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot red', () => {
    const { container } = render(<Button type="red" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot outlined', () => {
    const { container } = render(<Button type="outlined" label="test label" />);
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot transparent', () => {
    const { container } = render(
      <Button type="transparent" label="test label" />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot htmlType submit', () => {
    const { container } = render(
      <Button type="transparent" htmlType="submit" label="test label" />
    );
    expect(container).toMatchSnapshot();
  });

  it('display correct text passed by label', () => {
    const { container } = render(<Button type="primary" label="test label" />);
    expect(container.querySelector('button').innerHTML).toBe('test label');
  });
});
