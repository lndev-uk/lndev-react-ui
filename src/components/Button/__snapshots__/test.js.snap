// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot green 1`] = `
<div>
  <button
    class="button  button--green  "
  >
    test label
  </button>
</div>
`;

exports[`Button matches snapshot htmlType submit 1`] = `
<div>
  <button
    type="submit"
    class="button  button--transparent  "
  >
    test label
  </button>
</div>
`;

exports[`Button matches snapshot outlined 1`] = `
<div>
  <button
    class="button  button--outlined  "
  >
    test label
  </button>
</div>
`;

exports[`Button matches snapshot red 1`] = `
<div>
  <button
    class="button  button--red  "
  >
    test label
  </button>
</div>
`;

exports[`Button matches snapshot secondary 1`] = `
<div>
  <button
    class="button  button--secondary  "
  >
    test label
  </button>
</div>
`;

exports[`Button matches snapshot success 1`] = `
<div>
  <button
    class="button  button--primary  "
  >
    test label
  </button>
</div>
`;

exports[`Button matches snapshot transparent 1`] = `
<div>
  <button
    class="button  button--transparent  "
  >
    test label
  </button>
</div>
`;
